src/main/java/org/minions/devfund/chelo/WhackAMole.java

```diff
@@ -0,0 +1,259 @@
+package org.minions.devfund.chelo;
+
+
+import java.util.Random;
+import java.util.Scanner;
+
+/**
+ * A class for WackAMole task.
+ */
+public class WhackAMole {
+    private int score, molesLeft, attemptsLeft;
+    private char[][] moleGrid;
+
+    /**
+     * Constructor for the game, specifies total number of whacks allowed, and the grid dimension.
+     * @param numAttempts initial tries for the user.
+     * @param gridDimension size of the grid.
+     */
+    WhackAMole(int numAttempts, int gridDimension) {
+        this.moleGrid = new char[gridDimension][gridDimension];
+        this.attemptsLeft = numAttempts;
+        this.score = 0;
+        this.molesLeft = 0;
+        initializeGrid();
+    }
+
+    /**
+     * A method to initialize the grid with asterisks.
+     */
+    void initializeGrid() {
+        for (int i = 0; i < this.moleGrid.length; i++) {
+            for (int j = 0; j < this.moleGrid.length; j++) {
+                this.moleGrid[i][j] = '*';
+            }
+        }
+    }
+
+    /**
+     * A method to randomically put the moles in the grid.
+     * @param molesLeft remaining moles
+     * @param dimensionLimit dimension of the grid's limit
+     */
+    void placeTheMoles(int molesLeft, int dimensionLimit) {
+        int x, y;
+        Random rand = new Random();
+
+        while (molesLeft > 0) {
+            x = rand.nextInt(dimensionLimit);
+            y = rand.nextInt(dimensionLimit);
+            if (place(x, y)) {
+                this.moleGrid[x][y] = 'M';
+                molesLeft--;
+            }
+        }
+    }
+
+    /**
+     * Method given a location, returns if availability.
+     * @param x coordinate
+     * @param y coordinate
+     * @return status of the coordinate
+     */
+    boolean place(int x, int y) {
+        if (this.moleGrid[x][y] != 'M') {
+            this.moleGrid[x][y] = 'M';
+            this.molesLeft++;
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * A method given a location, take a whack at that location.
+     * If that location contains a mole, the score, number of attemptsLeft, and molesLeft is updated.
+     * If that location does not contain a mole, only attemptsLeft is updated.
+     * @param x coordinate
+     * @param y coordinate
+     */
+    void whack(int x, int y) {
+        if (moleGrid[x][y] == 'M') {
+            moleGrid[x][y] = 'W';
+            score++;
+            molesLeft--;
+            attemptsLeft--;
+        } else {
```

Comment:
this else is no longer necessary.
You can declare attemptsLeft--;
directly in the next line and remove it in the if conditional.